File: app/src/main/java/in/woloo/www/giftSubscription/model/GetGiftPlansResponse.java
package in.woloo.www.giftSubscription.model;

import java.util.List;
import com.google.gson.annotations.SerializedName;

public class GetGiftPlansResponse{

	@SerializedName("code")
	private int code;

	@SerializedName("data")
	private List<DataItem> data;

	@SerializedName("message")
	private String message;

	@SerializedName("status")
	private String status;

	public int getCode(){
		return code;
	}

	public List<DataItem> getData(){
		return data;
	}

	public String getMessage(){
		return message;
	}

	public String getStatus(){
		return status;
	}

	public class DataItem{

		@SerializedName("image")
		private String image;

		@SerializedName("backgroud_color")
		private String backgroudColor;

		@SerializedName("price_with_gst")
		private String priceWithGst;

		@SerializedName("description")
		private String description;

		@SerializedName("discount")
		private String discount;

		@SerializedName("created_at")
		private String createdAt;

		@SerializedName("shield_color")
		private String shieldColor;

		@SerializedName("is_voucher")
		private int isVoucher;

		@SerializedName("deleted_at")
		private Object deletedAt;

		@SerializedName("frequency")
		private String frequency;

		@SerializedName("is_recommended")
		private int isRecommended;

		@SerializedName("is_expired")
		private int isExpired;

		@SerializedName("apple_product_id")
		private Object appleProductId;

		@SerializedName("updated_at")
		private String updatedAt;

		@SerializedName("price")
		private String price;

		@SerializedName("name")
		private String name;

		@SerializedName("days")
		private int days;

		@SerializedName("currency")
		private String currency;

		@SerializedName("id")
		private int id;

		@SerializedName("before_discount_price")
		private Object beforeDiscountPrice;

		@SerializedName("plan_id")
		private String planId;

		@SerializedName("status")
		private int status;

		@SerializedName("start_at")
		private String start_at;

		@SerializedName("end_at")
		private String end_at;

		public String getImage(){
			return image;
		}

		public String getBackgroudColor(){
			return backgroudColor;
		}

		public String getPriceWithGst(){
			return priceWithGst;
		}

		public String getDescription(){
			return description;
		}

		public String getDiscount(){
			return discount;
		}

		public String getCreatedAt(){
			return createdAt;
		}

		public String getShieldColor(){
			return shieldColor;
		}

		public int getIsVoucher(){
			return isVoucher;
		}

		public Object getDeletedAt(){
			return deletedAt;
		}

		public String getFrequency(){
			return frequency;
		}

		public int getIsRecommended(){
			return isRecommended;
		}

		public int getIsExpired(){
			return isExpired;
		}

		public Object getAppleProductId(){
			return appleProductId;
		}

		public String getUpdatedAt(){
			return updatedAt;
		}

		public String getPrice(){
			return price;
		}

		public String getName(){
			return name;
		}

		public int getDays(){
			return days;
		}

		public String getCurrency(){
			return currency;
		}

		public int getId(){
			return id;
		}

		public Object getBeforeDiscountPrice(){
			return beforeDiscountPrice;
		}

		public String getPlanId(){
			return planId;
		}

		public int getStatus(){
			return status;
		}

		public String getStart_at() {
			return start_at;
		}

		public String getEnd_at() {
			return end_at;
		}
	}
}

File: app/src/main/java/in/woloo/www/more/giftSubscription/model/GetGiftPlansResponse.kt
package `in`.woloo.www.more.giftSubscription.model

import com.google.gson.annotations.SerializedName

class GetGiftPlansResponse {
    @SerializedName("code")
    val code: Int = 0

    @kotlin.jvm.JvmField
	@SerializedName("data")
    val data: List<DataItem>? = null

    @SerializedName("message")
    val message: String? = null

    @SerializedName("status")
    val status: String? = null

    inner class DataItem {
        @SerializedName("image")
        val image: String? = null

        @SerializedName("backgroud_color")
        val backgroudColor: String? = null

        @SerializedName("price_with_gst")
        val priceWithGst: String? = null

        @SerializedName("description")
        val description: String? = null

        @SerializedName("discount")
        val discount: String? = null

        @SerializedName("created_at")
        val createdAt: String? = null

        @SerializedName("shield_color")
        val shieldColor: String? = null

        @SerializedName("is_voucher")
        val isVoucher: Int = 0

        @SerializedName("deleted_at")
        val deletedAt: Any? = null

        @SerializedName("frequency")
        val frequency: String? = null

        @SerializedName("is_recommended")
        val isRecommended: Int = 0

        @SerializedName("is_expired")
        val isExpired: Int = 0

        @SerializedName("apple_product_id")
        val appleProductId: Any? = null

        @SerializedName("updated_at")
        val updatedAt: String? = null

        @SerializedName("price")
        val price: String? = null

        @SerializedName("name")
        val name: String? = null

        @SerializedName("days")
        val days: Int = 0

        @SerializedName("currency")
        val currency: String? = null

        @SerializedName("id")
        val id: Int = 0

        @SerializedName("before_discount_price")
        val beforeDiscountPrice: Any? = null

        @SerializedName("plan_id")
        val planId: String? = null

        @SerializedName("status")
        val status: Int = 0

        @SerializedName("start_at")
        val start_at: String? = null

        @SerializedName("end_at")
        val end_at: String? = null
    }
}